tests/services/label.service.spec.ts

```diff
@@ -1,9 +1,30 @@
 import { LabelService } from '../../src/app/core/services/label.service';
 import { Label } from '../../src/app/core/models/label.model';
 import * as LabelConstant from '../constants/label.constants';
+import { Observable, of } from 'rxjs';
 
 let labelService: LabelService;
 let labelList: Label[];
+let githubService: any;
+
+describe('LabelService', () => {
+    beforeEach(() => {
+      githubService = jasmine.createSpyObj('GithubService', ['fetchAllLabels', 'createLabel']);
+      labelService = new LabelService(githubService);
+    });
+
+    describe('.syncLabels()', () => {
+      it('should return the result of labelService.synchronizeRemoteLabels()', () => {
```

Comment:
This test currently doesn't verify any useful behaviour.

Let's rewrite this test to verify that label synchronisation occurs correctly.
This area of our app is currently untested.

Suggested steps:
- mock `githubService.fetchAllLabels` to return an array of raw label objects. This simulates the step where the app queries GitHub to find the labels that currently exist in the repo. We can use the existing test data available in `label.constants.ts` (in our `tests` code)
- In the subscriber callback, test that `githubService.createLabel` is called with correct arguments --> checks whether LabelService correctly requested for label creation via GithubService
- No need to check exhaustively that every required label is created; testing a few should be sufficient.

Reply:
Alright! I'll have a go at this, 

so we just need to check via `toHaveBeenCalledTimes` and `toHaveBeenCalledWith` wrt to `githubService.createLabel` am i right to say that o.o